sonar-application/src/main/java/org/sonar/application/Cluster.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.application;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.hazelcast.cluster.ClusterState;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.JoinConfig;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import java.util.Arrays;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Manager for the cluster communication between Main Processes
+ */
+public class Cluster implements AutoCloseable {
+  /**
+   * The Hazelcast instance.
+   */
+  @VisibleForTesting
+  public HazelcastInstance hazelcastInstance;
+
+  /**
+   * Instantiates a new Cluster.
+   *
+   * @param clusterProperties The properties of the cluster read from configuration
+   */
+  protected Cluster(ClusterProperties clusterProperties) {
+    if (clusterProperties.isEnabled()) {
+      initializeHazelcast(clusterProperties);
+    }
+  }
+
+  private void initializeHazelcast(ClusterProperties clusterProperties) {
```

Comment:
if this method is supposed to throw errors and that we are relying on them for validating the users's input, tests are missing.

that said, I think  some, if not all, of the validation should be done by sonarqube itself:

1. relying on an external framework (which point is not to do validation) to provide error message means these messages may change without notice during an version upgrade
2. most importantly, these messages won't mention the property the user must modify to fix the problem and therefor will be pretty much useless to the user